### PR TITLE
[Merged by Bors] - chore: `one_le_pow_of_one_le` -> `one_le_pow₀`

### PR DESCRIPTION
**Pull request by YaelDillies**
Rename:
* `pow_le_one` -> `pow_le_one₀`
* `pow_lt_one` -> `pow_lt_one₀`
* `one_le_pow_of_one_le` -> `one_le_pow₀`
* `one_lt_pow` -> `one_lt_pow₀`

and make all non-order arguments implicit (those are not rewrite lemmas, and where the explicit arguments are located is basically random throughout the library)

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Order/Monoid/Unbundled/Pow.lean [hunk 1]
```diff
@@ -52,9 +52,9 @@ theorem pow_lt_one_of_lt {a : M} {n : ℕ} (h : a < 1) (hn : n ≠ 0) : a ^ n <
 
 end Left
 
-@[to_additive nsmul_nonneg] alias one_le_pow_of_one_le' := Left.one_le_pow_of_le
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Unbundled/Pow.lean:55]: Deprecation for this one seems to be missing; didn't check the others

Comment (YaelDillies) [reply Mathlib/Algebra/Order/Monoid/Unbundled/Pow.lean]: This shouldn't have been in the diff, sorry

File: Mathlib/Algebra/Order/Ring/Basic.lean [hunk 2]
```diff
@@ -61,23 +61,31 @@ theorem pow_add_pow_le (hx : 0 ≤ x) (hy : 0 ≤ y) (hn : n ≠ 0) : x ^ n + y
         exact mul_le_mul_of_nonneg_left (ih (Nat.succ_ne_zero k)) h2
 
 @[bound]
-theorem pow_le_one : ∀ n : ℕ, 0 ≤ a → a ≤ 1 → a ^ n ≤ 1
+theorem pow_le_one₀ : ∀ {n : ℕ}, 0 ≤ a → a ≤ 1 → a ^ n ≤ 1
```
Comment (b-mehta) [inline Mathlib/Algebra/Order/Ring/Basic.lean:64]: This PR also changes the implicitness of some lemmas. In particular it makes `n` implicit in lemmas which do not have a hypothesis relating to `n`. Please note this change in the PR description, together with a justification for it.

**Review by b-mehta — CHANGES_REQUESTED** [review]

File: Mathlib/Algebra/Order/Ring/Basic.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Order/Ring/Basic.lean:64]: Done

Comment (b-mehta) [reply Mathlib/Algebra/Order/Ring/Basic.lean:64]: Can you explain what you mean by "where the explicit arguments are located is basically random throughout the library"? As mentioned above, locally the location of the explicit argument is pretty clear and consistent, so I don't follow the argument here.

Comment (b-mehta) [reply Mathlib/Algebra/Order/Ring/Basic.lean:64]: I'm not questioning swapping the order - that part is fine - it's the change in implicitness that I'm concerned about. The other lemmas in this PR take `n` as an implicit argument exactly when it is mentioned in one of the other hypotheses, showing consistency. As such, the PR description should justify why the implicitness has been changed.

Comment (b-mehta) [reply Mathlib/Algebra/Order/Ring/Basic.lean:64]: Mm okay, that's convincing.

**Review by b-mehta — COMMENTED** [review]
Looks good to me, but I'd like someone else to check too.

maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**